gpflow/kernels/base.py

```diff
@@ -119,19 +119,19 @@ def slice_cov(self, cov: tf.Tensor) -> tf.Tensor:
 
     @abc.abstractmethod
     def K(self, X, Y=None, presliced=False):
-        pass
+        raise NotImplementedError
 
     @abc.abstractmethod
     def K_diag(self, X, presliced=False):
-        pass
+        raise NotImplementedError
 
     def __call__(self, X, Y=None, full=True, presliced=False):
         if not full and Y is not None:
             raise ValueError(
                 "Ambiguous inputs: `diagonal` and `y` are not compatible.")
         if not full:
-            return self.K_diag(X, presliced=presliced)
-        return self.K(X, Y, presliced=presliced)
```

Comment:
What's the intent behind no longer passing through the `presliced` argument ?

Reply:
`presliced` was only used for certain bits in the kernel expectation quadrature. In correspondence with @awav and @vincent we decided to simplify the kernel expectations. A consequence of this is that we will remove certain quadrature fallback behaviour that required the `presliced` behaviour. This simplifies kernels as well.

I will remove all `presliced` references in a separate PR.